### PR TITLE
Fix some form

### PR DESCRIPTION
**Pull request by lomamech**
Improve validation and invalidation of form and fix error

### CODE REVIEW

File: client/src/partials/cash/cashbox_account_currency/cashbox_account_currency.html [hunk 1]
```diff
@@ -70,84 +70,84 @@
...
 
         <div class="form-group">
           <label class="control-label">{{ "CASH.CASHBOX.ACCOUNT_VIREMENT" | translate }}</label>
-          <select class="form-control" ng-model="session.edit.virement_account_id" ng-options="p.id as formatAccount(p) for p in account1.data">
+          <select class="form-control" required ng-model="session.edit.virement_account_id" ng-options="p.id as formatAccount(p) for p in account1.data">
           </select>
         </div>
 
-        <button class="btn btn-sm btn-success" type="submit" ng-click="save.edit()">{{ "FORM.SUBMIT" | translate }}</button>
+        <button class="btn btn-sm btn-success" ng-disabled="edit.$invalid" type="submit" ng-click="save.edit()">{{ "FORM.SUBMIT" | translate }}</button>
```
Comment (jniles) [inline client/src/partials/cash/cashbox_account_currency/cashbox_account_currency.html]: Awesome work!  This is exactly the way form validation should work in Angular!

File: client/src/partials/grade_employers/grade_employers.html [hunk 2]
```diff
@@ -61,66 +61,66 @@
...
+              <input ng-model="session.edit.text" class="form-bhima" required>
             </div>
           </div>
 
           <div class="form-group">
             <label class="control-label col-xs-3 required">{{ "COLUMNS.BASIC_SALARY" | translate }}</label>
             <div class="input-group col-xs-8">
-              <input ng-model="session.edit.basic_salary" class="form-bhima" required="required">
+              <input ng-model="session.edit.basic_salary" class="form-bhima" required>
               <span class="input-group-addon"> {{ enterprise.data[0].symbol }} </span>
```
Comment (jniles) [inline client/src/partials/grade_employers/grade_employers.html]: Is there a safer way of doing this?  Can we use `appstate.register('enterprise')` to get the proper enterprise?

File: client/src/partials/reference/reference.html [hunk 3]
```diff
@@ -123,34 +123,34 @@
...
-              <input ng-model="session.edit.text" class="form-bhima" required="required">
+              <input ng-model="session.edit.text" class="form-bhima" required>
             </div>
           </div>
 
           <div class="form-group">
             <label class="control-label col-xs-4 required">{{ "COLUMNS.POSITION" | translate }}</label>
             <div class="col-xs-7">
-              <input ng-model="session.edit.position" class="form-bhima" required="required">
+              <input type="number" ng-model="session.edit.position" class="form-bhima" required>
```
Comment (jniles) [inline client/src/partials/reference/reference.html]: Would this be enhanced by placing a "min" and "max" value?

File: client/src/partials/reference_group/reference_group.html [hunk 4]
```diff
@@ -63,27 +63,27 @@
...
-              <input ng-model="session.new.text" class="form-bhima" required="required">
+              <input ng-model="session.new.text" class="form-bhima" required>
             </div>
           </div>
 
           <div class="form-group">
             <label class="control-label col-xs-4 required">{{ "COLUMNS.POSITION" | translate }}</label>
             <div class="col-xs-7">
-              <input ng-model="session.new.position" class="form-bhima" required="required">
+              <input type="number" ng-model="session.new.position" class="form-bhima" required>
```
Comment (jniles) [inline client/src/partials/reference_group/reference_group.html]: Would this benefit from a "min" property?

File: client/src/partials/reference_group/reference_group.html [hunk 5]
```diff
@@ -127,21 +127,24 @@
...
             </div>
           </div>
 
           <div class="form-group">
             <label class="control-label col-xs-4 required">{{ "COLUMNS.BALANCE_SECTION" | translate }}</label>
             <div class="col-xs-7">
-              <select class="form-bhima" ng-model="session.edit.section_bilan_id" ng-options="section.id as section.text  for section in section_bilan.data | orderBy:'text'">
-                <option>--- {{ "SELECT.BALANCE_SECTION" | translate }} ---</option>
+              <select class="form-bhima" ng-model="session.edit.section_bilan_id" required> 
+                <option value="" disabled="disabled"></option>
```
Comment (jniles) [inline client/src/partials/reference_group/reference_group.html]: Should there be a default option?

File: client/src/partials/reference_group/reference_group.js [hunk 6]
```diff
@@ -102,21 +96,15 @@ angular.module('bhima.controllers')
...
+      .then(function (res) {
+        messenger.success($translate.instant('REFERENCE_GROUP.SAVE_SUCCES'));  
+        record.id = res.data.insertId;               
         $scope.reference_groups.post(record);
         session.action = '';
         session.new = {};
       });
     };
 
-    function generateReference () {
```
Comment (jniles) [inline client/src/partials/reference_group/reference_group.js]: I am happy to see these functions being removed.  They were a terrible hack up until now. :)

**Comment by jniles** [comment]
Minor edits, then LGTM.  There is some really good work in this PR.

File: client/src/partials/reference/reference.html (hunk 3 above)
Comment (DedrickEnc) [inline client/src/partials/reference/reference.html]: Yes, we can put zero as a min value

Comment (DedrickEnc) [inline client/src/partials/reference/reference.html]: Yes, we can put zero as a min value but not the max

**Pull request merged**